src/components/JournalPage/index.jsx

```diff
@@ -43,21 +44,21 @@ class JournalPage extends React.Component {
                     // rich text and raw HTML use the same component
                     return <RawHTMLViewer content={el.value} />;
                   case PDF:
-                    return (<PDFViewer
+                    return (<ViewerBorder
                       spanId={el.value.span_id}
-                      url={el.value.url}
                       title={el.value.title}
+                      spanFullWidth
+                      child={<PDFViewer url={el.value.url} title={el.value.title} />}
                     />);
                   case XBLOCK_VIDEO:
                     // return <XBlockVideo />
                     return <div>XBlockVideo</div>;
                   case IMAGE:
-                    return (<ImageViewer
-                      url={el.value.url}
+                    return (<ViewerBorder
```

Comment:
This actually seems like a good use case for `props.children` to make it look more like nested components:
`
<ViewerBorder spanId=.....>
   <ImageViewer url=.. />
</ViewerBorder>
`
You would then just need to change `{props.child}` to `{props.children}` in your ViewerBorder render method. `props.children` gets passed in automagically

Reply:
That is super neat! do i still define it the same way in ```ViewerBorder.propTypes```?
Would it be: ```children: PropTypes.element.isRequired``` ?

Reply:
Yup! Exactly. You could even use `PropTypes.oneOfType` if you want to be explicit about the 3 types of elements you support